test/e2e/service.go

```diff
@@ -859,13 +869,19 @@ spec:
 			framework.Failf("Unable to write CRD config to disk: %v", err)
 		}
 		framework.Logf("Create the Load Balancer configuration")
-		framework.RunKubectlOrDie("default", "create", "-f", loadBalancerYaml)
+		e2ekubectl.RunKubectlOrDie("default", "create", "-f", loadBalancerYaml)
 
 	})
 
 	ginkgo.AfterEach(func() {
+		if ginkgo.CurrentGinkgoTestDescription().Failed {
+			dumpBGPInfo(reportPath, ginkgo.CurrentSpecReport().LeafNodeText, f)
```

Comment:
Are you aware of the `wrappedTestFramework` utility?

https://github.com/ovn-org/ovn-kubernetes/blob/ff456a740b501bd43687364c89adf5800d995dbd/test/e2e/util.go#L901

Would it be reasonable to hook this into that function? Maybe to collect the test namespace by default and optionally provide additional namespaces to collect?

Reply:
Needed bgp specific configuration and logs from metallb-system and test namespace pods only for lb related tests failure, so just added this to be part of these tests. We can move into wrappedTestFramework if there is a generic need for all test cases.